src/sentry/uptime/models.py

```diff
@@ -418,4 +445,18 @@ def create_detector_from_project_subscription(project_sub: ProjectUptimeSubscrip
     )
     DataSourceDetector.objects.create(data_source=data_source, detector=detector)
 
+    # Create DetectorState based on the uptime_status from the uptime_subscription
+    if project_sub.uptime_subscription.uptime_status == UptimeStatus.FAILED:
+        DetectorState.objects.create(
+            detector=detector,
+            state=DetectorPriorityLevel.HIGH,
+            is_triggered=True,
+        )
+    else:
+        DetectorState.objects.create(
+            detector=detector,
+            state=DetectorPriorityLevel.OK,
+            is_triggered=False,
+        )
```

Comment:
Why do we create the state here? Isn't this just happening on detector creation? So presumably the state is not failed?